widgets/src/main/java/org/pentaho/gwt/widgets/client/dialogs/DialogBox.java

```diff
@@ -48,7 +54,105 @@ public DialogBox( boolean autoHide, boolean modal ) {
     this.modal = modal;
     addPopupListener( this );
     setStylePrimaryName( "pentaho-dialog" );
+
+    // ARIA
+    setAriaRole( (Role) null );
+
+    setDomModal( this.isModal() );
+
+    // aria-labelledby
+    Widget labelWidget = getCaption().asWidget();
+    Roles.getHeadingRole().set( labelWidget.getElement() );
+    Roles.getHeadingRole().setAriaLevelProperty( labelWidget.getElement(), 2 );
+
+    Roles.getDialogRole().setAriaLabelledbyProperty( getElement(), Id.of( ensureId( labelWidget ) ) );
+
+    // Layout
+    // This should be the contained decorator panel's layout table.
+    // div > div > tab
+    Roles.getPresentationRole().set( getElement().getFirstChildElement().getFirstChildElement() );
+  }
+
+  // region aria role property
+  /**
+   * Gets the ARIA role of the dialog.
+   */
+  public String getAriaRole() {
+    return getElement().getAttribute( "role" );
+  }
+
+  /**
+   * Sets the ARIA role of the dialog given as a string.
+   *
+   * @param ariaRole The new ARIA <code>role</code> attribute.
+   *                 When <code>null</code> or empty, it is set to the default value of <code>dialog</code>.
+   */
+  public void setAriaRole( String ariaRole ) {
+    if ( StringUtils.isEmpty( ariaRole ) ) {
+      ariaRole = "dialog";
+    }
+
+    getElement().setAttribute( "role", ariaRole );
+  }
+
+  /**
+   * Sets the ARIA role of the dialog given as a {@link Role} instance.
+   *
+   * @param ariaRole The new ARIA <code>role</code> attribute.
+   *                 When <code>null</code>, it is set to the default value of {@link Roles#getDialogRole()}.
+   */
+  protected void setAriaRole( Role ariaRole ) {
```

Comment:
the method placement in a class should be public goes first and then protected and then default and then private